src/loaders/gtdb/compute_genome_attribs.py

```diff
@@ -0,0 +1,369 @@
+
+    # retrieve genomic.fna.gz files
+    fna_files = list()
+    for genome_id in genome_ids:
+        genome_files = _find_genome_file(genome_id, 'genomic.fna.gz', source_data_dir)
+        # Only need xx.genomic.fna.gz file.
+        # Excluding potential xx.cds_from_genomic.fna.gz and xx.rna_from_genomic.fna.gz files
+        genome_file = [f for f in genome_files if
+                       all(name_part not in f for name_part in ['cds_from', 'rna_from', 'ERR'])]
+        if genome_file and len(genome_file) == 1:
+            fna_files.append(genome_file[0])
+        else:
+            print(f'Cannot retrieve target file. Please check download folder for genome: {genome_id}')
+
+    start = time.time()
+
+    # RUN checkM2 predict
+    command = ['checkm2', 'predict',
+               '--output-directory', batch_dir,
+               '--threads', str(program_threads),
+               '--force',
+               '--input']
+    command.extend(fna_files)
+    command.append('--debug') if debug else None
+    print(f'running {" ".join(command)}')
+    _run_command(command, debug=debug, log_dir=os.path.join(batch_dir, 'checkm2_log'))
+    end_time = time.time()
+    print(
+        f'Used {round((end_time - start) / 60, 2)} minutes to execute checkM2 predict for {len(genome_ids)} genomes')
+
+    # TODO: inspect stdout for failed ids or do it in the parser program
+    return failed_ids
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='PROTOTYPE - Compute genome attributes in addition to GTDB metadata.')
+    required = parser.add_argument_group('required named arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    # Required flag arguments
+    required.add_argument('--tools', required=True, type=str, nargs='+',
+                          help='Tools to be executed. (e.g. gtdb_tk, checkm2, etc.)')
+    required.add_argument('--load_ver', required=True, type=str,
+                          help='KBase load version. (e.g. r207.kbase.1)')
+    required.add_argument('--source_data_dir', required=True, type=str,
+                          help='Source data (genome files) directory. '
+                               '(e.g. /global/cfs/cdirs/kbase/collections/sourcedata/GTDB/r207')
+
+    # Optional arguments
+    optional.add_argument('--kbase_collection', type=str, default='GTDB',
+                          help='KBase collection identifier name. (default: GTDB)')
+    optional.add_argument('--root_dir', type=str, default=ROOT_DIR,
+                          help='Root directory.')
+    optional.add_argument('--threads', type=int,
+                          help='Total number of threads. (default: half of system cpu count)')
+    optional.add_argument('--program_threads', type=int, default=32,
+                          help='Number of threads for each program execution. (default: 32)')
+    optional.add_argument('--node_id', type=str, default=str(uuid.uuid4()),
```

Comment:
Yes, it's used when we eventually run this script with multiple nodes. I did this because different nodes can overwrite other nodes' result dir if we only have `batch_x_size_x`.

Reply:
:+1: